hosting/views.py

```diff
@@ -948,41 +964,52 @@ def post(self, request, *args, **kwargs):
             email=owner.email,
             password=owner.password
         )
-        vm_data = VirtualMachineSerializer(manager.get_vm(vm.id)).data
-        terminated = manager.delete_vm(
-            vm.id
-        )
 
-        if not terminated:
-            messages.error(
-                request,
-                'Error terminating VM %s' % (opennebula_vm_id)
+        try:
+            vm_data = VirtualMachineSerializer(manager.get_vm(vm.id)).data
+        except BaseException:
```

Comment:
When/why do we except a BaseException here? Somewhat unclear to me and why do redirect user to the virutal_machines url.

Reply:
this is to handle the case, when a user tries to delete an a;ready deleted VM. THis can happen if I open two different pages for a particular VM and terminate the VM from one page, and once the termination is completed, then try to terminate it from the other page also.

Reply:
Thanks for the explanation, but in that case, should we not expect a `WrongIdError` - BaseException sounds too generic to me..

For the second point, I guess the kwargs parameter does not make any sense in that case.

Reply:
yes, good points. Updating accordingly.